helsingborg-stad/app-mitt-helsingborg
# Bug/subform validation

## Explain the changes you’ve made

In accordance with #CU-1exphyp, I've made changes to not let users save subforms without entering values.

## Explain why these changes are made

As users shouldn't save a field without entering a value, these changes are made.

## Explain your solution

In order to facilitate checking the subform fields in the easiest way, I've added a way to know when a user has added a new answer to a `RepeaterField` and added a handler of that event which sets the `RepeaterField` to dirty and performs a validation on it.

As the user shouldn't be able to save the subform with empty values, I've changed the `backToMain` case in StepFooter to also check for errors before allowing the user to move on.

In this pull request, I've also made the `RepeaterField` answers show their validation errors.

## How to test the changes?

Concrete example:
1. Checkout this branch
2. Start a case for "Ensamsökande"
3. Try adding an empty field in "medicin och sjukvård".
4. See that it doesn't let you

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.


## Anything else?

In order for this to work, we're going to need some changes to be made using the form builder.
More specifically, we need to set the inputs rows on repeater fields to have required answers.
